chpl/chpl-api/src/main/java/gov/healthit/chpl/listener/ProductNamesCacheRefreshListener.java

```diff
@@ -0,0 +1,73 @@
+package gov.healthit.chpl.listener;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.aspectj.lang.annotation.AfterReturning;
+import org.aspectj.lang.annotation.Aspect;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import gov.healthit.chpl.caching.CacheRefreshListener;
+import gov.healthit.chpl.caching.updater.ProductNamesCacheUpdater;
+import gov.healthit.chpl.domain.PendingCertifiedProductDetails;
+import gov.healthit.chpl.domain.SplitProductsRequest;
+import gov.healthit.chpl.domain.UpdateProductsRequest;
+
+/**
+ * Listener that determines when to update the product names cache.
+ * @author kekey
+ *
+ */
+@Component
+@Aspect
+public class ProductNamesCacheRefreshListener extends CacheRefreshListener {
+    private static final Logger LOGGER = LogManager.getLogger(ProductNamesCacheRefreshListener.class);
+    @Autowired
```

Comment:
We'd talked about moving these autowired things to constructors. Is that something that can be done here?